### PR TITLE
DAOS-3934 build: Remove ompi from build

### PR DESCRIPTION
**Pull request by jolivier23**
Update to latest scons_local
Install scons_local/env_modules.py to get load_mpi for tests
Update to latest cart
Update daos tests to use installed openmpi and to switch
dynamically between mpich and openmpi as needed
Update go tests to add ISAL path
Fix an issue with client ignoring -DER_EVICTED (DAOS-3932)

Signed-off-by: Jeff Olivier <jeffrey.v.olivier@intel.com>

### CODE REVIEW

File: utils/docker/Dockerfile.leap.15 [hunk 1]
```diff
@@ -27,7 +27,8 @@ RUN zypper --non-interactive --no-gpg-checks refresh; \
            libuuid-devel libyaml-devel                                \
            make man meson nasm ninja pandoc patch python2-pip         \
            readline-devel sg3_utils which yasm                        \
-	   python-devel python3-devel valgrind-devel
+	   python-devel python3-devel valgrind-devel hwloc-devel          \
```
Comment (brianjmurrell) [inline utils/docker/Dockerfile.leap.15]: Looks like you have some extra whitespace before the `\` at the end of this line.

File: utils/docker/Dockerfile.leap.15 [hunk 2]
```diff
@@ -27,7 +27,8 @@ RUN zypper --non-interactive --no-gpg-checks refresh; \
...
+       openmpi3-devel Modules
```
Comment (brianjmurrell) [inline utils/docker/Dockerfile.leap.15]: And not enough at the beginning of this line.

File: utils/rpms/daos.spec [hunk 3]
```diff
@@ -333,6 +335,9 @@ getent group daos_admins >/dev/null || groupadd -r daos_admins
 %{_libdir}/*.a
 
 %changelog
+* Fri Dec 13 2019 Jeff Olivier <jeffrey.v.olivier@intel.com> - 0.7.0-2
+- Remove openmpi, pmix, and hwloc build requirement, use packages
```
Comment (brianjmurrell) [inline utils/rpms/daos.spec]: Adding `hwloc`[`-devel`] build requirement here though, aren't we?

**Comment by brianjmurrell** [comment]
I should have added in my review comment, nothing here seems to be anything in particular to -1 for.  Just a bunch of comments and suggestions.  Some stuff that could be fixed if you needed to push again for any reason.

File: Jenkinsfile [hunk 4]
```diff
@@ -429,7 +430,7 @@ pipeline {
                             */
                         }
                         unsuccessful {
-                            sh 'ls install/include/spdk/ install/include/hwloc.h || true'
+                            sh 'ls install/include/spdk/ || true'
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply Jenkinsfile]: done

File: src/tests/daos_perf.sh [hunk 5]
```diff
@@ -34,7 +34,7 @@ if (( $# >= 4 )); then	# -s $RSIZE
 	RSIZE=$4	#    value size
 fi
 
-ORTERUN="${OMPI_PATH}/bin/orterun"
+ORTERUN=$(which orterun)
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply src/tests/daos_perf.sh]: done

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 6]
```diff
@@ -115,16 +117,20 @@ class TestWithoutServers(Test):
     def setUp(self):
         """Set up run before each test."""
         super(TestWithoutServers, self).setUp()
+        load_mpi('openmpi')
+        self.orterun = find_executable('orterun')
+        if self.orterun is None:
+            self.fail("Could not find orterun")
+
+        self.ompi_prefix = os.path.dirname(os.path.dirname(self.orterun))
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply src/tests/ftest/util/apricot/apricot/test.py]: Indeed.  I was just trying to keep the changes minimal but @phender and I discussed simplification

File: src/tests/ftest/util/daos_perf_utils.py [hunk 7]
```diff
@@ -140,9 +143,13 @@ def get_launch_command(self, basepath, processes, hostfile, runpath=None):
...
 
+        load_mpi('openmpi')
+        orterun_bin = find_executable('orterun')
+        if orterun_bin is None:
+            raise DaosPerfFailed("orterun not found")
+
         orterun_cmd = [
-            os.path.join(runpath if runpath else build_paths[
-                "OMPI_PREFIX"], "bin/orterun"),
+            orterun_bin,
```
[the remark this answers is not in the document]
Comment (jolivier23) [reply src/tests/ftest/util/daos_perf_utils.py]: same answer

File: utils/docker/Dockerfile.leap.15 (hunk 1 above)
Comment (jolivier23) [reply utils/docker/Dockerfile.leap.15]: done

File: utils/rpms/daos.spec (hunk 3 above)
Comment (jolivier23) [reply utils/rpms/daos.spec]: I modified the comment

File: utils/docker/Dockerfile.leap.15 [hunk 8]
```diff
@@ -27,7 +27,8 @@ RUN zypper --non-interactive --no-gpg-checks refresh; \
            libuuid-devel libyaml-devel                                \
            make man meson nasm ninja pandoc patch python2-pip         \
            readline-devel sg3_utils which yasm                        \
-	   python-devel python3-devel valgrind-devel
+	       python-devel python3-devel valgrind-devel hwloc-devel      \
```
Comment (frostedcmos) [inline utils/docker/Dockerfile.leap.15]: seems visually misaligned

File: src/tests/ftest/util/apricot/apricot/test.py [hunk 9]
```diff
@@ -115,16 +118,25 @@ class TestWithoutServers(Test):
...
         """Set up run before each test."""
         super(TestWithoutServers, self).setUp()
+        load_mpi('openmpi')
+        self.orterun = find_executable('orterun')
+        if self.orterun is None:
+            self.fail("Could not find orterun")
+
+        # hardware tests segfault in MPI_Init without this option
+        self.client_mca = "--mca btl_openib_warn_default_gid_prefix 0"
+        self.client_mca += " --mca pml ob1"
```
Comment (frostedcmos) [inline src/tests/ftest/util/apricot/apricot/test.py]: Just a note that these options might be different for non-sockets providers

File: src/tests/ftest/util/server_utils.py [hunk 10]
```diff
@@ -652,12 +654,16 @@ def run_server(test, hostfile, setname, uri_path=None, env_dict=None,
...
-            os.path.join(build_vars["OMPI_PREFIX"], "bin", "orterun"),
-            "--np", str(server_count)]
-        if uri_path is not None:
-            server_cmd.extend(["--report-uri", uri_path])
+        load_mpi('openmpi')
+        orterun_bin = find_executable('orterun')
+        if orterun_bin is None:
+            raise ServerFailed("Can't find orterun")
+
+        server_cmd = [orterun_bin, "--np", str(server_count)]
```
Comment (frostedcmos) [inline src/tests/ftest/util/server_utils.py]: Same note as above

**Pull request merged**